plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -383,6 +382,25 @@ public void addEnvironmentVariable(String key, String value) {
 		this.environmentVariables.add(new EnvironmentVariable(key, value, true));
 		firePropertyChange(PROPERTY_ENVIRONMENT_VARIABLES, old, Collections.unmodifiableList(environmentVariables));
 	}
+	
+	
+
+	@Override
+	public void addServicePort(IServicePort port) {
+		List<IServicePort> old = new ArrayList<>(this.servicePorts);
```

Comment:
Technically yes even though the calling code has protections.  We could modify it to depend on Collections and utilize a Set?